wordpress-mobile/WordPress-Android
# Added onRetryUploadForMediaClicked interface for GB

Introduces `Retry/Retry all/Remove` for failed media uploads in Gutenberg

gutenberg-mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/528
gutenberg PR https://github.com/WordPress/gutenberg/pull/13615


#### To test:
CASE A RETRY (single)
1. start a new draft
2. insert image block
3. choose DEVICE LIBRARY
4. choose an image (better a big one or throttle your connection to make it last)
5. once it starts uploading, turn airplane mode ON
6. observe it errors and displays `Failed to insert media.\nPlease tap for options.`
7. turn airplane mode OFF
8. tap on the image
9. observe the dialog appears with options REMOVE/RETRY ALL/ RETRY
10. tap on RETRY
11. observe the image starts to upload again, and finishes successfully.

![testa](https://user-images.githubusercontent.com/6597771/52025685-07972480-24e4-11e9-8e4a-08c153badf54.gif)



CASE B RETRY (multiple)
1. start a new draft
2. insert image block
3. insert another image block
4. choose DEVICE LIBRARY on one of them
5. choose an image (better a big one or throttle your connection to make it last - I used 2G as preferred network and turned wifi off to make it worse)
6. once it starts uploading, tap DEVICE LIBRARY on the other block and do the same
7. once you see both are slowly uploading, turn airplane mode ON.
8. observe it errors and displays `Failed to insert media.\nPlease tap for options.`
9. turn airplane mode OFF
10. tap on the image
11. observe the dialog appears with options REMOVE/RETRY ALL/ RETRY
12. tap on RETRY ALL
13. observe both images start to upload again

![testb2](https://user-images.githubusercontent.com/6597771/52025796-7b393180-24e4-11e9-97c7-0511625afacb.gif)


CASE C REMOVE
1. repeat steps 1-9 of case A
2. tap on REMOVE
3. observe the block is reset as not containing any image.

![testc](https://user-images.githubusercontent.com/6597771/52025715-24cbf300-24e4-11e9-95c7-ba1552572e70.gif)

CASE D: cancel
1. start a draft
2. insert an image block
3. choose an image from the device
4. while it's uploading, tap on it and observe the  "Stop uploading?" dialog appears.
5. tap YES to cancel upload, or NO to dismiss the dialog.

![testd](https://user-images.githubusercontent.com/6597771/52027687-0fa69280-24eb-11e9-975d-4859bce70b1c.gif)


Update release notes:
